builtins/xrootd/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.
+# All rights reserved.
+#
+# For the licensing terms see $ROOTSYS/LICENSE.
+# For the list of contributors see $ROOTSYS/README/CREDITS.
+
+include(ExternalProject)
+
+find_package(OpenSSL REQUIRED)
+
+set(XROOTD_VERSION "5.1.0")
+set(XROOTD_VERSIONNUM 500010000 CACHE INTERNAL "" FORCE)
+set(XROOTD_SRC_URI https://github.com/xrootd/xrootd/archive/v${XROOTD_VERSION}.tar.gz)
+set(XROOTD_DESTDIR ${CMAKE_BINARY_DIR}/XROOTD-prefix)
+set(XROOTD_ROOTDIR ${XROOTD_DESTDIR})
+message(STATUS "Downloading and building XROOTD version ${XROOTD_VERSION}")
+
+  # Guess under which directory XRootD will install its libraires
+  set(XROOTD_LIBDIR "lib")
+  if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND ${CMAKE_SIZEOF_VOID_P} EQUAL 8
+     AND NOT CMAKE_CROSSCOMPILING AND NOT EXISTS "/etc/debian_version")
+    set(XROOTD_LIBDIR "lib64")
+  endif()
+
+  set(XROOTD_LIBRARIES ${XROOTD_ROOTDIR}/${XROOTD_LIBDIR}/libXrdUtils${CMAKE_SHARED_LIBRARY_SUFFIX}
+                       ${XROOTD_ROOTDIR}/${XROOTD_LIBDIR}/libXrdClient${CMAKE_SHARED_LIBRARY_SUFFIX}
```

Comment:
@oshadura 
libXrdClient no longer provided in the xrootd 5.x,
If I rermove it here, I can compile ROOT on my system.
Version 5.1 is working for me.
But we have to check that `net/netx` disabled for xrootd 5.x while `net/netx` based on client library

Reply:
Thanks a lot for feedback! I will add your suggestion.

>>But we have to check that net/netx disabled for xrootd 5.x while net/netx based on client library
it is already done some time ago.

Reply:
> it is already done some time ago.

Unfortunately not - if I specify `-Dnetx=ON` ROOT will try to build it without checking of XROOTD version.
If you want, I can add these fixes to your PR

Reply:
hmmm, I thought https://github.com/root-project/root/blob/master/cmake/modules/SearchInstalledSoftware.cmake#L980-L989 will work, let me try to build with `-Dnetx=ON`.

Reply:
I got problem when I call cmake second time adding `-Dnetx=ON`.
But I have anyway problems with incremental builds now, but this is another issue

Reply:
And now you move setting of xrootd version number in other cmake file - is it a problem?
I propose, that these checks done directly in `net/netx` and `net/netxng` - at the moment when they build

Reply:
Thanks, let me try to experiment with your suggestion.